Test/SearchEngineTests.cs

```diff
@@ -628,6 +628,110 @@ public static void TestSemiSpecificEnzymeEngineSingleC()
             Assert.AreEqual("QQQGGGG", allPsmsArray[0].BaseSequence);
         }
 
+        [Test]
+        public static void TestSemiTrypsin()
+        {
+            var myMsDataFile = new TestDataFile("Yes, I'd like one slightly larger please");
+            var variableModifications = new List<ModificationWithMass>();
+            var fixedModifications = new List<ModificationWithMass>();
+            var localizeableModifications = new List<ModificationWithMass>();
+            Dictionary<ModificationWithMass, ushort> modsDictionary = new Dictionary<ModificationWithMass, ushort>();
+            foreach (var mod in fixedModifications)
+                modsDictionary.Add(mod, 0);
+            int ii = 1;
+            foreach (var mod in variableModifications)
+            {
+                modsDictionary.Add(mod, (ushort)ii);
+                ii++;
+            }
+            foreach (var mod in localizeableModifications)
+            {
+                modsDictionary.Add(mod, (ushort)ii);
+                ii++;
+            }
+
+            var proteinList = new List<Protein> { new Protein("MGGGGGMKNNNQQQGGGGKGKKNKKGN", "hello") };
```

Comment:
Please throw in a proteolysis product with start/end indices NOT corresponding to lysines/arginines. And make sure the fragments are generated accordingly.